test/forge/ReallocateWithdrawTest.sol

```diff
@@ -122,9 +144,25 @@ contract ReallocateWithdrawTest is IntegrationTest {
             suppliedAssets[2].toSharesDown(totalSupplyAssets[2], totalSupplyShares[2])
         ];
 
-        if (suppliedShares[0] > 0) supplied.push(MarketAllocation(allMarkets[0], suppliedAssets[0], 0));
-        if (suppliedAssets[1] > 0) supplied.push(MarketAllocation(allMarkets[1], 0, suppliedShares[1]));
-        if (suppliedShares[2] > 0) supplied.push(MarketAllocation(allMarkets[2], suppliedAssets[2], 0));
+        if (suppliedShares[0] > 0) {
+            supplied.push(MarketAllocation(allMarkets[0], suppliedAssets[0], 0));
+            vm.expectEmit();
+            emit EventsLib.ReallocateSupply(allMarkets[0].id(), suppliedAssets[0]);
+        }
+        if (suppliedAssets[1] > 0) {
+            supplied.push(MarketAllocation(allMarkets[1], 0, suppliedShares[1]));
+            vm.expectEmit();
+            emit EventsLib.ReallocateSupply(allMarkets[1].id(), suppliedAssets[1]);
+        }
+        if (suppliedShares[2] > 0) {
+            supplied.push(MarketAllocation(allMarkets[2], suppliedAssets[2], 0));
+            vm.expectEmit();
+            emit EventsLib.ReallocateSupply(allMarkets[2].id(), suppliedAssets[2]);
+        }
+
+        vm.expectEmit();
+        if (expectedIdle > formerIdle) emit EventsLib.ReallocateIdle(expectedIdle - formerIdle, 0);
```

Comment:
worth to add 2 different tests to be sure always testing the 2 branches invoking ReallocateIdle

Reply:
Hmm if it's not tested then fuzzing is broken... 😅

Reply:
Mmmh not really. For example, this test can never be triggered by fuzzer:

```solidity
function testSomething(uint256 generatedByFuzzer) external {
  if(generatedByFuzzer == 13534656856383547)
    // Never tested code
  else // Tested code
}
```

In my understanding of the fuzzer, generated values are not generated by checking the comparison in tests.

I agree with you that, in this specific case, it's probably tested, but a golden rule in software testing is never to add if..else in a test (given... when... expect...)

Reply:
I agree for the rule but I just don't think it's worth spending time on duplicating or refactoring the code but I'll do it

Reply:
In fact I won't, it will be removed

Reply:
> In fact I won't, it will be removed

Let's wait for the conclusion of the idle removal, and revert on it if not implemented.